hypervisor/dm/vpci/vroot_port.c

```diff
@@ -140,14 +141,19 @@ int32_t create_vrp(struct acrn_vm *vm, struct acrn_vdev *dev)
 			dev_config->vdev_ops = &vrp_ops;
 
 			vdev = vpci_init_vdev(&vm->vpci, dev_config, NULL);
+			if (vdev == NULL) {
+				pr_err("%s: failed to create virtual root port\n", __func__);
+				ret = -EFAULT;
+				break;
+			}
 
 			init_ptm(vdev, vrp_config);
 
 			break;
 		}
 	}
 
-	return 0;
+	return ret;
 }
 
 int32_t destroy_vrp(__unused struct pci_vdev *vdev)
```

Comment:
Need call vpci_deinit_vdev() in destroy_vrp().

Reply:
Fixed in 2nd commit